packages/eslint-plugin-fiori-tools/src/language/annotations/source-code.ts

```diff
@@ -0,0 +1,163 @@
+        //         for (const subElement of child.subElements) {
+        //             if (subElement.type === 'XMLElement' && subElement.name === 'Include') {
+        //                 let namespace: string | undefined;
+        //                 let alias: string | undefined;
+        //                 for (const attribute of subElement.attributes) {
+        //                     if (attribute.key === 'Namespace') {
+        //                         namespace = attribute.value ?? undefined;
+        //                     } else if (attribute.key === 'Alias') {
+        //                         alias = attribute.value ?? undefined;
+        //                     }
+        //                 }
+        //                 if (namespace) {
+        //                     this.aliasMap[namespace] = namespace;
+        //                     if (alias) {
+        //                         this.aliasMap[alias] = namespace;
+        //                     }
+        //                 }
+        //             }
+        //         }
+        //     }
+        // }
+        return this.aliasMap;
+    }
+
+    // getParent(node: XMLAstNode | XMLToken): object | undefined {
+    //     if (isNode(node)) {
+    //         if (node.type === 'XMLDocument') {
+    //             return undefined;
+    //         }
+    //         return node.parent;
+    //     } else {
+    //         return undefined;
+    //     }
+    // }
+
+    /**
+     *
+     * @returns
+     */
+    traverse(): Iterable<TraversalStep> {
+        const steps: TraversalStep[] = [];
+        const visit = (node: AnyNode, parent?: AnyNode) => {
+            steps.push(
+                new AnnotationTraversalStep({
+                    target: node,
+                    phase: STEP_PHASE.ENTER,
+                    args: [node, parent]
+                })
+            );
+
+            for (const key of visitorKeys[node.type] ?? []) {
+                // @ts-expect-error 7053
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                const child = node[key];
+
+                if (child) {
+                    if (Array.isArray(child)) {
+                        child.forEach((grandchild) => {
+                            visit(grandchild, node);
+                        });
+                    } else {
+                        visit(child, node);
+                    }
+                }
+            }
+            steps.push(
+                new AnnotationTraversalStep({
+                    target: node,
+                    phase: STEP_PHASE.EXIT,
+                    args: [node, parent]
+                })
+            );
+        };
+        visit(this.ast, undefined);
+
+        return steps;
+    }
+
+    /**
+     * Get location of node or token.
+     *
+     * @param node - Node or token
+     * @returns Location
+     */
+    getLoc(node: AnyNode): SourceLocation {
```

Comment:
As per eslint doc https://eslint.org/docs/latest/extend/languages#the-sourcecode-object we should implement `getRange`? too. Is this inherited?

There are some recommendations too
- lines
- getParent
- getAncestors
- getText

Maybe future improvements.